File: 2015/08/01.py
import ast

with open("input.txt", "r") as file:
    lines = file.read().strip().splitlines()

total_code_chars = sum(len(line) for line in lines)
total_memory_chars = sum(len(ast.literal_eval(line)) for line in lines)
print(f"Part 01 : {total_code_chars - total_memory_chars}")

File: 2025/01/01.lua
local file = io.open("input.txt", "r")
local dial, count = 50, 0

for line in file:lines() do
    local n = tonumber(line:sub(2)) % 100
    dial = (line:sub(1, 1) == "L" and dial - n or dial + n) % 100
    count = count + ((dial == 0) and 1 or 0)
end

print("Part 01 : " .. count)